Dodanie plików do głównego repo

krzycho2 opened the pull request.

krzycho2 committed:
Add files via upload

Algorytm typu Region growing by gray values udoskonalony przez dodanie progowania Otsu.

krzycho2 committed:
Merge pull request #1 from krzycho2/krzycho2-patch-1

Add files via upload

krzycho2 committed:
Pierwszy commit

krzycho2 committed:
Kolejny

krzycho2 committed:
Delete nowy.py

krzycho2 committed:
Poprawnie działająca średnia i odchylenie standardowe

krzycho2 committed:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector

krzycho2 committed:
Usunięcie starej wersji, wszystko jest w nowej.

krzycho2 committed:
Właściwie nic nowego

krzycho2 committed:
Lepiej wyglądający histogram

krzycho2 committed:
Delete Algorytm_region_growing.py

krzycho2 committed:
Delete nowy.py

krzycho2 committed:
Add files via upload

krzycho2 committed:
Wszystko

krzycho2 committed:
Skrypt wczytujący fantoma

krzycho2 committed:
Merge branch 'master' into Porg

krzycho2 committed:
Merge pull request #2 from krzycho2/Porg

Porg

krzycho2 committed:
Delete fantom.m

krzycho2 committed:
Delete make_csr

krzycho2 committed:
Delete LICENSE

krzycho2 committed:
Delete ForwardedMessage.eml

krzycho2 committed:
dodany plik progowanie

krzycho2 committed:
Porządki

krzycho2 committed:
Porządki

krzycho2 committed:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector

krzycho2 committed:
Porządki cd

Commit:
Dodanie nowego artykułu

Commit:
Porządki dalsze

Commit:
dodany zapis zmiennych do słoika

Commit:
Segmentacja ręczna

Commit:
Stworzona klasa do obsługi danych

krzycho2 committed:
Drobne poprawki

Commit:
Poprawki w VolumeData

Commit:
Poprawki w klasie

Commit:
Ładniegit status

Commit:
Koniec progowania

Commit:
Reszta

Commit:
Poprawki w progowaniu ręcznym

krzycho2 committed:
Create README.md

krzycho2 committed:
Update README.md

Commit:
jest okMerge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector

Commit:
Porządkowanie tego bałaganu

Commit:
Poprawki czytelności wczytywania nowych rekonstrukcji

krzycho2 committed:
Delete read_data.py

Nieużywane.

krzycho2 committed:
Delete Histogram.png

Commit:
Dodanie opisu klasy VolumeData

Commit:
Usunięcie artykułów

Commit:
Porządki

krzycho2 committed:
Update README.md

krzycho2 committed:
Delete test2.txt

krzycho2 committed:
Delete test.py

Commit:
Poprawienie wczytywania obrazów

Commit:
Poprawienie wczytywania obrazów

Commit:
Poprawki sumaryczne

Commit:
Stworzenie metody do segmentacji obrazów na podstawie podanych progów

Commit:
Testowanie progowania Otsu, na podstawie implementacji Scikit Image

Commit:
Fajniejsza metoda segmentacji

Commit:
Poprawki w segmentowaniu danych

Commit:
Poprawki w konstruktorze

Commit:
Przeniesienie tekstów powitalnych (main) oraz pozostałych funkcji do lib.py

Commit:
Stworzenie nowego pliku do obsługi plików/zmiennych z danymi do volumeData.py

Commit:
Uporządkowanie działanie konstruktora. Użycie gotowych narzędzi do wczytywania danych od numpy. Działanie na uint8 - oszczędniej.

Commit:
Poprawki w interfejsie - dodane są komunikaty o błędach.

Commit:
Zrezygnowanie z kolorów. Kolorowanie nie iteracyjne, a przy pomocy wyrażeń logicznych - ogromny wzrost szybkości działania programu

Commit:
Skończony algorytm Yena

Commit:
Przeniesienie pozostałych metod

Commit:
Dołączenie metod progowania do klasy

Commit:
Finalizacja

Commit:
Poprawki

krzycho2 committed:
Delete algorytmy.py

krzycho2 committed:
Delete seg-new-data.py

krzycho2 committed:
Delete temp.py

krzycho2 committed:
Update README.md

krzycho2 committed:
Update README.md

krzycho2 committed:
Update README.md

Commit:
a

Commit:
Prawie gotowy region growing

Commit:
Poprawki

graag commented on Inne/pakiety.txt:
```diff
@@ -0,0 +1,10 @@
+numpy
```
Przenieś do requirements.txt

https://pip.pypa.io/en/stable/user_guide/#requirements-files

krzycho2 replied:
Ok.

graag commented on SegmentujObrazyJPET.py:
```diff
@@ -0,0 +1,106 @@
+    elif alg == 'otsu-iter':
+        segImage = image3D.otsu_iterSegment()
+    elif alg == 'otsu-region':
+        image3D.otsu_regionSegment()
```
Nie zwracasz segImage?

graag commented on SegmentujObrazyJPET.py:
```diff
@@ -0,0 +1,106 @@
+    # Pokazanie przekrojów po segmentacji
+    slices = segImage.showAllSlicesInOne(title =f'Wynik segmentacji algorytmem  {alg}')
+    
+    return slices
```
Nie lepiej zwrócić cały segImage?

I zdecydować potem czy chcesz go pokazać czy zapisać do pliku do dalszej obróbki?

krzycho2 replied:
Ok

graag commented on SegmentujObrazyJPET.py:
```diff
@@ -0,0 +1,106 @@
+# PROGRAM #
+
+# Program wywołany bez argumentów
+argList = sys.argv
```
Mam mieszane odczucia.

Wolałbym implementację opartą o argparse i bez wersji interaktywnej. Z tego co pamiętam `input` nie obsługuje uzupełniania nazw plików żeby było użyteczne by trzeba użyć jakiegoś readline.

Ale bardziej przydatny będzie działający region growing i otsu wielopoziomowy ...

krzycho2 replied:
Skoro bez wersji interaktywnej to lepiej:
a) stworzyć program wywołany z wiersza poleceń - python Segmentuj[...].py , czyli tak jak jest tylko bez interkacji z userem czy
b) udostępnić projekt jako API?

krzycho2 replied:
Skoro bez wersji interaktywnej to lepiej:
a) stworzyć program wywołany z wiersza poleceń - python Segmentuj[...].py , czyli tak jak jest tylko bez interkacji z userem czy
b) udostępnić projekt jako API?

graag replied:
A możemy oba?

W tym sensie żeby to z czego korzysta CLI które napisałeś było biblioteką z używalnym API i miało odpowiednią dokumentację (wystarczą docstringi z opisem działania funkcji jej  parametrów i tego co zwraca return - najlepiej w formacie sphinx-a)

krzycho2 replied:
Gotowe

graag commented on adaptive-thresh.py:
```diff
@@ -0,0 +1,39 @@
+from lib import VolumeData
```
Planujesz to używać? Jeśli nie to przenieś do podkatalogu jakiegoś. Np `misc`

krzycho2 replied:
Usunięte.

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+            licz += 1
+    return img
+
+krotkieInfo = """
```
Ponieważ to są stałe to jednak by je jako takie oznaczył np: `KROTKIE_INFO`

krzycho2 replied:
Ok.

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+
+def getListOfFiles(dirName):
+    # create a list of file and sub directories 
+    # names in the given directory 
```
https://docs.python.org/3/library/os.html#os.walk

graag replied:
Jakoś tak:

```
import os
allFiles = []
for root, dirs, files in os.walk(dirName):
    for name in files:
        allFiles.append(os.path.join(root, name))
```

krzycho2 replied:
Ale co jest w sumie z tym nie tak ?

graag replied:
Owszem działa ale to nie jest zadanie napisania algorytmu przechodzącego drzewo katalogów a implementacji programu który będzie miał relatywnie mało bug-ów.

Nie ma powodu żeby pisać tego typu rzeczy od zera kiedy są dostępne rozwiązania w stdlib. Na pewno są lepiej przetestowane i lepiej radzą sobie z corner case-ami.

Raz że nie masz obsługi błędów którą w os.walk można łatwo dodać podając coś parametrowi onerror lub ignorować błędy poszczególnych katalogów bez jej definiowania.

Dwa że jest szybsze:
> Changed in version 3.5: This function now calls os.scandir() instead of os.listdir(), making it faster by reducing the number of calls to os.stat().

krzycho2 replied:
Ok, poprawione.

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+    for i in range(Nx):
+        for j in range(Ny):
+            for k in range(Nz):
+                img3D[Ny-1-j,i,k] = rawData[licz,3]          # Przekopiowanie danych
```
Czemu taka kolejność współrzędnych y,x,z?

krzycho2 replied:
Wytłumaczę dla 2D.
Chcemy przedstawić obraz punktów za pomocą macierzy:
x    y     f(x,y)
x0  y0     0.1
x0  y1     0.2
x1  y0     0.3
x1  y1     0.4
Gdybyśmy nie zamienili indeksów to macierz (obraz) wyglądałaby (po obróćeniu względem osi ox) o tak: 
   _y_____
x | 0.1   0.2
   | 0.3   0.4
w tym momencie pionowo(rzędna) są x, a poziomo(odcięta) y, co jest błędne - oczekujemy, żęby rzędną był y, a odciętą x.

Dodatkowo trzeba obrócić macierz/obraz względem osi odciętych.

graag replied:
Oczywiście. Jakoś zapomniałem że pierwszy indeks to wiersz a drugi kolumna.

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+    return allFiles
+
+def arr2img(arr):
+    """
```
To jest jeszcze używane?

krzycho2 replied:
Wszystkie "narzędzia" zakładają, że dane wejściowe mogą być również w 2D. Chyba, że z tego zrezygnować...

graag replied:
OK i to założenie jest poprawne. Ale nadal nie widzę gdzie używasz `arr2img`:

```
konrad@nb53cis ~/Soft/GitHub/j-pet-roi-detector $ grep -r arr2img
Inne/prog-basic.py:from lib import VolumeData, vol2Img3D, arr2img
lib.py:def arr2img(arr):
```

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+    w0k, w1k, w2k   = 0,0,0       # Jakieś wagi, double
+    m0k, m1k, n2k   = 0,0,0       # Średnie 
+    m0, m1, m2      = 0,0,0
+    m0k, m1k, mt    = 0,0,0
```
Czemu redefiniujesz zmienne?

graag commented on lib.py:
```diff
@@ -0,0 +1,228 @@
+    hist = np.zeros(maxVal)
+    for i in range(h):
+        for j in range(w):
+            val = slice75[i,j]
```
Chyba powinieneś użyć cały obrazek a nie tylko slice, Nie?

graag replied:
I chyba możesz skorzystać z `np.histogram`

krzycho2 replied:
Zapraszam do najnowszej wersji :)

graag commented on multi-thresh.py:
```diff
@@ -0,0 +1,96 @@
+import numpy as np
```
Do usunięcia lub przeniesienia. Zgadza się?

krzycho2 replied:
Zgadza się - do usunięcia.

graag commented on volumeData.py:
```diff
@@ -0,0 +1,269 @@
+        self._data3D = rawData
+        
+
+    def savePickle(self, filePath = __name__ + 'Data.pckl'):    
```
Lepszy jako domyślny filepatch byłby albo `os.getcwd` albo katalog pliku z danymi czyli z _dataName (o ile jako dane wejściowe nie zostało użyte np.array)

graag commented on volumeData.py:
```diff
@@ -0,0 +1,269 @@
+        Nz = self._data3D.shape[2]   # Liczba przekrojów
+        maxValue = np.amax(self._data3D)
+        minValue = np.amin(self._data3D)
+        rows = 6
```
Pewnie mogłobybyć `rows = sqrt(Nz)` albo `sqrt(Nz)/(4/3)`

graag commented on volumeData.py:
```diff
@@ -0,0 +1,269 @@
+            ax.imshow(self.getSlice(sliceNum=index), vmin=minValue, vmax=maxValue)
+            ax.set_title( index )
+        
+        plt.show()
```
Możliwość zapisania do pliku?

krzycho2 replied:
Jest możliwość zapisania do pliku pickle. To wygodny format, moim zdaniem to wystarczy.

krzycho2 replied:
Jest możliwość zapisania do pliku pickle. To wygodny format, moim zdaniem to wystarczy.

graag replied:
Myślałem o zapisie plotów do png.

graag commented on regionGrowing.py:
```diff
@@ -0,0 +1,192 @@
+def sigma(arr, thresh):
+    """Odchylenie standardowe liczb powyżej progu"""
+    temp = arr - srednia(arr, thresh)
+    var = srednia((arr - srednia(arr,thresh))**2, thresh)
```
1. Ponieważ będziesz potrzebował najczęściej i sredniej i sigma to powinna byc jedna funkcja ktora zwraca tuplę (mean, sigma) - wtedy nie liczysz sredniej 2x
2. Ta linia jest bledna bo zewnetrzna srednia nalozy progowanie na wartosci kwadratow odchylen an nie na wartosci macierzy

Proponowalbym najpierw wyznaczyc macierz po nalozeniu progu i zapisać jako MaskedArray: https://docs.scipy.org/doc/numpy/reference/maskedarray.generic.html#constructing-masked-arrays

Następnie użyć:
- https://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.mean.html#numpy.ma.mean

Niestety numpy nie ma funkcji która będzie zwracać jednocześnie mean i std. Pozostaje coś w tym stylu:

```
m = np.ma.mean(a)
s = np.ma.mean((a-m)**2)
```

krzycho2 replied:
A mógłbyś zajrzeć do najnowszej wersji tego algorytmu ? Jest on teraz metodą klasy VolumeData. Ponieważ próg, który przekazujemy do tego algorytmu miał służyć do wybrania woksela startowego. Kolejne woksele mają być dodawane do regionu, jeśli spełniają równanie: woksel - średnia <= c*sigma - bez względu na to, czy są większe niż próg czy mniejsze IMHO. W najnowszej wersji algorytmu średnią i sigmę liczę jako numpy.mean oraz sqrt(numpy.var).

graag commented on regionGrowing.py:
```diff
@@ -0,0 +1,192 @@
+        return np.sqrt(var)
+
+def pointsInRadius(sPoint, radius, dims=2):
+    points = []
```
Jeśli szukamy punktów w okręgu to tu jest fajna implementacja:

https://stackoverflow.com/questions/44865023/circular-masking-an-image-in-python-using-numpy-arrays

powinna się dobrze przenieść na 3 wymiary. Dostaniesz maskę którą można użyć do MaskedArray

graag commented on regionGrowing.py:
```diff
@@ -0,0 +1,192 @@
+
+centroidy = np.array(centroidy)
+startPoints = np.around(centroidy).astype(int)
+startPoint= startPoints[2]
```
Do tego miejsca wygląda OK

Czemu `startPoints[2]` ??

Co do samego algorytmu to chyba nie ma sensu ograniczać się do wyszukiwania w coraz rosnącym promieniu:
- Będzie ciężko stwierdzić które punkty powinny należeć do regionu a które nie (które są połączone)
- Dla regionów które nie są symetryczne będziemy przemiatali sporo zbednych punktów

Najlepiej chyba użyć jakiejś implementacji `dilate` np: https://scikit-image.org/docs/dev/api/skimage.morphology.html#skimage.morphology.dilation

Dostaniesz region powiększony np o jeden piksel w każdym kierunku. Wtedy moższ nałożyć na niego progowanie. Po progowaniu wyznaczyć średnią i odchylenie używając maski. Następnie powtórzyć algorytm.

krzycho2 replied:
Chyba nie za bardzo rozumiem idei. W tym momencie tworzony obszar trzymam jako listę. Idąc po spirali dodaję kolejne punkty do listy w zależności od tego czy spełniają równanie czy nie. Na jakim obiekcie miałbym wykonać dylację ? Czy chodzi o to, żeby stworzyć pustą macierz o wymiarach badanego wolumenu i na niej wykonywać dylację, a punkty dodane przez dylację następnie przepuszczać przez równanie ? To mi się wydaje o wiele bardziej nieekonomiczne pamięciowo, ponieważ mielibyśmy pustą w 99% macierz 160x160x120. 
Tymczasem, gdy używam punktów z coraz większego promienia, to mam tam warunke stopu: gdy żaden z punktów w promieniu R nie zostanie dodany do obszaru, algorytm przerywa działanie. I w ten sposób maksymalny promień jaki udało mi się osiągnąć dla fantoma to 4, czyli 64 wartości.

krzycho2 committed:
Delete seg-new-data.py

krzycho2 committed:
Delete adaptive-thresh.py

Commit:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector

Wszystko jet ok

Commit:
Przeniesienie implementacji algorytmów do VolumeData

Commit:
Przeniesienie funkcji do modułu lib

krzycho2 committed:
Delete seg-otsu.py

Zawartość przeniesiona do pliku lib.py

krzycho2 committed:
Delete seg-prog.py

Te algorytmy nie są wykorzystywane - nie sprawdziły się. Lepszą implementację progowania zapewnia scikit-image.

Commit:
Dodanie pliku z wymaganymi pakietami.

Commit:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector
It's nessesary.

Commit:
Dodanie progowania otsu wielowartościowego.

Commit:
Nazwy stałych wielkimi literami.

Commit:
Dopasowanie nazw stałych

krzycho2 committed:
Delete multi-thresh.py

Algorytm przeniesiony do pliku lib.py.

Commit:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector
Jest ok

krzycho2 committed:
Delete pakiety.txt

Wszystko jest w requirements.txt

Commit:
Porządkowanie

krzycho2 committed:
Delete regionGrowing.py

Implementacja algorytmu przeniesiona do pliku VolumeData.py (metody klasy VolumeData), a funkcji pomocniczych do lib.py.

Commit:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector
ok

Commit:
Dodanie angielskiej wersji językowej.

krzycho2 committed:
Update README.md

Commit:
Dodanie implementacji progowania dwuwartościowego Otsu

krzycho2 committed:
Update README.md

krzycho2 committed:
Delete prog-basic.py

Commit:
Dodanie do repo progowania ręcznego oraz spisu treści pracy inżynierskiej

Commit:
Merge branch 'master' of https://github.com/krzycho2/j-pet-roi-detector
fajen

graag commented on volumeData.py:
```diff
@@ -0,0 +1,357 @@
+                print(f'Promień: {R}')
+                points = lib.pointsInRadius(startPoint, R)
+                if not lib.belongsToArr(points, img):    # Sprawdzenie, czy przynajmniej jeden punkt należy do macierzy
+                    print('Wyjście poza macierz wszystkich punktów.')
```
Jak już tak chcesz to należałoby porównać startPoint +/- R z rozmiarem img.

Wszystkie punkty będą poza obrazem tylko jak start_x+R > rozmiarX( img ), start_y+R > rozmiarY(img) oraz start_x-R<0, start_y-R<0. To wynika z tego że twój nowy region jest kwadratem, nie?

graag commented on volumeData.py:
```diff
@@ -0,0 +1,357 @@
+                        print(f'tempWynik: {tempWynik}, sigmaC: {sigmaC}')
+
+                        if tempWynik <= sigmaC:
+                            region.append(p)
```
Wadą tego algorytmu jest to że potrafi generować obszary które są nieciągłe. Dla przykładu:
```
000010000
000010000
000010000
000010000
000000000
000000000
000000000
000010000
```

Startując z czwartego rzędu i 5 kolumny algorytm złapie wszystkie jedynki.

O ile czyni go to porównywalnym z progowaniem Otsu czy innym to tracimy zaletę region growing która powinna ograniczyć przestrzeń wyszukiwania do obszaru ciągłego.

Przy bardzo dużych obrazach i małych regionach można by pomyśleć o macierzach rzadkich. Ale z tego co widzę może to wymagać więcej pracy niż to warto.

Myślę że wystarczy pracować na masce binarnej:
- zaczynamy od maski gdzie wszystkie elementy są `False` poza punktem startowym
- na niej wykonujemy operację `skimage.morphology.binary_dilation`
- można porównać z poprzednim obrazem żeby dostać maskę nowych punktów
- dla nowych punktów które wyciągamy ich indeksy i sprawdzamy czy ich intensywność nie różni się średniej o więcej niż sigma. Usuwamy z maski te które mają większą różnicę
- żeby efektywnie liczyć średnią można wykorzystać twoją implementację z zbiorem wartości intensywności punktów. W pierwszym kroku wyciągamy indeksy z maski binarnej dla wartości True w drugim kroku indeksujemy obraz i wyciągamy wartości intensywności.

Commit:
Poprawiona metoda getListOfFiles

Commit:
Utworzenie klasy SegmentVolume, która będzie odpowiadać konkretnie za segmentację. Klasa VolumeData będzie przechowywać dane wolumenu

Commit:
Nowe podawanie argumentow - przez argparse

Commit:
Poprawki w wyświetlaniu sliceów. Zapis sliceów do pliku.

Commit:
Wszystkie algorytmy przeniesione tutaj

Commit:
Poprawki

Commit:
Poprawki

Commit:
Poprawienie interfejsu CLI użytkownika. Wykorzystanie argparse.

Commit:
Komunikacja z użytkownikiem z wykorzystaniem argparse.

Commit:
Komunikacja z użytkownikiem z wykorzystaniem argparse.

Commit:
Przeniesienie algorytmów do klasy segmentVolume - enkapsulacja.

graag commented on lib.py:
```diff
@@ -176,19 +176,13 @@ def points2Img3D(rawData):
+    Tworzy listę wszystkich plików w danej lokalizacji
+    """
+    allFiles = []
+    for root, dirs, files in os.walk('/home/krzysztof/Pulpit/Praca_inż'):
```
dirName zamiast '/home/krzysztof/Pulpit/Praca_inż' ??

krzycho2 replied:
Jasne

graag commented on segmentVolume.py:
```diff
@@ -0,0 +1,345 @@
+        self.__algName = algName
+        print(f'Segmentacja algorytmem {algName}.')
+        if params:
+            self.__segmentedVolume = self.__algorithms[algName](params)
```
Fajny pomysł tylko nie wszystkie algorytmy spełniają taki interfejs.

Przy okazji czy params przekazany przez argparse nie będzie string-iem. Widzę że w części algorytmów oczekujesz tablicy ...

Może niech params będzie odczytanym z json-a słownikiem który przekarzesz tutaj jako `**params`. Słownik wtedy może zawierać klucze które odpowieadają nazwanym argumentom funkcji które implementują algorytmy?

Przy obsłudze argparse zawartość `args.algParams` wrzucasz w `json.loads`.

krzycho2 replied:
argparse.add_paramater(argName, **type=json.loads**) 
Kurcze ile ja się męczyłem, żeby to znaleźć... 
Ostatecznie działa.

graag reviewed (commented):
Projekt wygląda już bardzo dobrze.

Podoba mi się rozdzielenie zadań między VolumeData i SegmentVolume.
Trochę mało konsekwentne jest umieszczenie segmentacji Otsu w lib skoro pozostałe algorytmy są bezpośrednio w SegmentVolume.

Do rozwiązania jest algorytm region grownig. Jeśli masz sposób na zapewnienie ciągłości regionów inny niż `dilation` z zachowaniem obecnej implementacji też może być pod warunkiem że nie będzie zbyt złożony algorytmicznie/obliczeniowo.

Pozostały drobiazgi takie jak zapisana na sztywno ścieżka, nie jasne wykorzystanie arr2img i obsługa parametrów algorytmu.

Commit:
Przeniesienie wszystkich algorytmów progujących do segmentVolume. Enkapsulacja. Usunięcie metody arr2img, bo nieużywana.

Commit:
Przeniesienie wszystkich algorytmów progujących do segmentVolume.

Commit:
Dopasowanie argparse, aby przyjmował argumenty pozycyjne i podawał je jako kwargs wgłąb programu.

Commit:
Bez większych zmian.

Commit:
Nowy region growing. Argumenty do algorytmów przekazuje się teraz przez słownik.

Commit:
nic nowego

Commit:
Poprawienie podawania parametrów. Czyszczenie kodu.

graag commented on segmentVolume.py:
```diff
@@ -99,155 +99,176 @@ def segmentation(self, algName, params=None):
-        region_c, region_sigma0 = params
+    def yenThreshRegionSegmentation(self, **params):
+        """
+        Explanatione
```
<LOL>

graag reviewed (approved):
Jest OK. Zrobię merge.

Przeleć te printy bo mam wrażenie że zostało sporo debug-ów.

Przejrzę jeszcze jak wygląda całość i ewentualnie dorzucę jakieś uwagi.
